docs/DeveloperGuide.adoc

```diff
@@ -337,46 +397,232 @@ Use case ends.
 * 3a. The given index is invalid.
 +
 [none]
-** 3a1. AddressBook shows an error message.
+** 3a1. MoneyGoWhere shows an error message.
 +
 Use case resumes at step 2.
 
-_{More to be added}_
+'''
+[discrete]
+=== UC03 Delete a spending
 
-[appendix]
-== Non Functional Requirements
+*MSS*
 
-.  Should work on any <<mainstream-os,mainstream OS>> as long as it has Java `11` or above installed.
-.  Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
-.  A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
+1.  User requests to view the list of spending.
+2.  MoneyGoWhere shows a list of spending.
+3.  User requests to delete a specific spending from the list.
+4.  MoneyGoWhere deletes the spending.
++
+Use case ends.
 
-_{More to be added}_
+*Extensions*
 
-[appendix]
-== Glossary
+[none]
+* 2a. The list is empty.
++
+Use case ends.
 
-[[mainstream-os]] Mainstream OS::
-Windows, Linux, Unix, OS-X
+* 3a. The given index is invalid.
++
+[none]
+** 3a1. MoneyGoWhere shows an error message.
++
+Use case resumes at step 2.
 
-[[private-contact-detail]] Private contact detail::
-A contact detail that is not meant to be shared with others
+'''
+[discrete]
+=== UC04 Find a spending
 
-[appendix]
-== Product Survey
+*MSS*
 
-*Product Name*
+1.  User requests to find all occurrences of a keyword or within a cost range or date range in a spending.
+2.  MoneyGoWhere returns any spending found with the keyword or within a cost range or date range specified.
++
+Use case ends.
 
-Author: ...
+'''
+[discrete]
+=== UC05 Set a monthly budget
+
+*MSS*
 
-Pros:
+1.  User requests to set a monthly budget
+2.  MoneyGoWhere sets a specific monthly budget
++
+Use case ends.
 
-* ...
-* ...
+*Extensions*
 
-Cons:
+[none]
+* 1a. The given budget value is invalid.
++
+[none]
+** 1a1. MoneyGoWhere shows an error message.
++
+Use case resumes at step 1.
+[none]
 
-* ...
-* ...
+'''
+[discrete]
+=== UC06 Generate statistics
+
+*MSS*
+
+1.  User requests to generate statistics.
+2.  MoneyGoWhere generates statistical analysis such as total spending of each tag.
++
+Use case ends.
+[none]
+
+'''
+[discrete]
+=== UC07 List all tags
+
+*MSS*
+
+1.  User requests to list all the recorded tags.
+2.  MoneyGoWhere lists all the recorded tags.
++
+Use case ends.
+[none]
+
+'''
+[discrete]
+=== UC08 Import external data from a CSV file
+
+*MSS*
+
+1.  User requests to import data from CSV file by specifying the file path.
+2.  MoneyGoWhere imports the data from the CSV file located at specified file path.
```

Comment:
"file path" might be a bit too technical for a use case. Maybe the specification could be worded as "User requests to import data from a CSV file" instead.